string patternName
float32 confidence
